mrp_lot_number_propagation/models/mrp_production.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2022 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class MrpProduction(models.Model):
+    _inherit = "mrp.production"
+
+    is_lot_number_propagated = fields.Boolean(
+        related="bom_id.lot_number_propagation",
+        readonly=True,
+        help=(
+            "Lot/serial number is propagated "
+            "from a component to the finished product."
+        ),
+    )
+    propagated_lot_producing = fields.Char(
+        compute="_compute_propagated_lot_producing",
+        help=(
+            "The BoM used on this manufacturing order is set to propagate "
+            "lot number from one of its components. The value will be "
+            "computed once the corresponding component is selected."
+        ),
+    )
+
+    @api.depends(
+        "move_raw_ids.bom_line_id.propagate_lot_number",
```

Comment:
Same as for the comment above, I feel like this should not be related to the BOM. Can we instead add a readonly `propagate_lot_number` Boolean field on `stock.move` and set it through `action_confirm`?